contracts/child/ChildToken/ChildBurnableERC1155.sol

```diff
@@ -0,0 +1,182 @@
+     * Make sure minting is done only by this function
+     * @param user user address for whom deposit is being done
+     * @param depositData abi encoded ids array and amounts array
+     */
+    function deposit(address user, bytes calldata depositData)
+        external
+        override
+        only(DEPOSITOR_ROLE)
+    {
+        (
+            uint256[] memory ids,
+            uint256[] memory amounts,
+            bytes memory data
+        ) = abi.decode(depositData, (uint256[], uint256[], bytes));
+        require(user != address(0x0), "ChildBurnableERC1155: INVALID_DEPOSIT_USER");
+        _mintBatch(user, ids, amounts, data);
+    }
+
+    /**
+     * @notice called when user wants to withdraw single token back to root chain
+     * @dev Should burn user's tokens. This transaction will be verified when exiting on root chain
+     * @param id id to withdraw
+     * @param amount amount to withdraw
+     */
+    function withdrawSingle(uint256 id, uint256 amount) external {
+        _burn(_msgSender(), id, amount);
+    }
+
+    /**
+     * @notice called when user wants to batch withdraw tokens back to root chain
+     * @dev Should burn user's tokens. This transaction will be verified when exiting on root chain
+     * @param ids ids to withdraw
+     * @param amounts amounts to withdraw
+     */
+    function withdrawBatch(uint256[] calldata ids, uint256[] calldata amounts)
+        external
+    {
+        _burnBatch(_msgSender(), ids, amounts);
+    }
+
+    /**
+     * Consider using this method, when you want to withdraw with some arbitrary metadata
+     */
+    function withdrawSingleWithMetadata(uint256 id, uint256 amount) external {
+        
+        emit TransferSingleWithMetadata(_msgSender(), _msgSender(), address(0), id, amount, encodeMetadata(_msgSender(), id, amount));
+
+        _burn(_msgSender(), id, amount);
+
+    }
+
+    /**
+     * Consider using this method, when you want to withdraw a batch of tokens, with some arbitrary metadata
+     */
+    function withdrawBatchWithMetadata(uint256[] calldata ids, uint256[] calldata amounts) external {
+        
+        emit TransferBatchWithMetadata(_msgSender(), _msgSender(), address(0), ids, amounts, encodeMetadata(_msgSender(), ids, amounts));
+
+        _burnBatch(_msgSender(), ids, amounts);
+
+    }
+
+    /**
+     * Consider using this method, when you want to **burn** on both L2, L1
+     */
+    function burnSingle(address user, uint256 id, uint256 amount) external only(DEFAULT_ADMIN_ROLE) {
+
+        _burn(user, id, amount);
+        emit BurnSingle(_msgSender(), user, id, amount);
+
+    }
+
+    /**
+     * Consider using this method, when you want to **burn** a batch of tokens, on both L2, L1
+     */
+    function burnBatch(address user, uint256[] calldata ids, uint256[] calldata amounts) external only(DEFAULT_ADMIN_ROLE) {
+
+        _burnBatch(user, ids, amounts);
+        emit BurnBatch(_msgSender(), user, ids, amounts);
```

Comment:
The same comment I have on ERC20 applies here

Having a ChainExit event would be better: for ERC1155 this would be something like:

`ChainExit(uint256 chainId, address to, uint256 tokenId, uint256 amount, bytes data)`

Note the `bytes data` is very important so extra data can be passed through, this can include the tokenURI for example

Reply:
POS bridge does have only two participating chains, which is why, `chainId` is not required. Because one can only exit from L2, not from L1.

Reply:
yes, do not worry about chainId, the point remains, ChainExit event is in my opinion a better solution that the Burn event as it can  even pass data